introduction_to_applying_machine_learning/deepar_chicago_traffic_violations/deepar_chicago_traffic_violations.ipynb

```diff
@@ -0,0 +1,467 @@
+    "    def __encode_request(self, ts, cat, encoding, num_samples, quantiles):\n",
+    "        instances = [series_to_obj(ts[k], cat[k] if cat else None) for k in range(len(ts))]\n",
+    "        configuration = {'num_samples': num_samples, 'output_types': ['quantiles'], 'quantiles': quantiles}\n",
+    "        http_request_data = {'instances': instances, 'configuration': configuration}\n",
+    "        return json.dumps(http_request_data).encode(encoding)\n",
+    "    \n",
+    "    def __decode_response(self, response, prediction_times, encoding):\n",
+    "        response_data = json.loads(response.decode(encoding))\n",
+    "        list_of_df = []\n",
+    "        for k in range(len(prediction_times)):\n",
+    "            prediction_index = pd.DatetimeIndex(start=prediction_times[k], freq=self.freq, periods=self.prediction_length)\n",
+    "            list_of_df.append(pd.DataFrame(data=response_data['predictions'][k]['quantiles'], index=prediction_index))\n",
+    "        return list_of_df\n",
+    "\n",
+    "\n",
+    "predictor = DeepARPredictor(endpoint=best_tuning_job_name,\n",
+    "                            sagemaker_session=sagemaker_session,\n",
+    "                            content_type=\"application/json\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we can use the previously created `predictor` object. We will predict only the first few time series, and compare the results with the actual data we kept in the test set."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predictor.set_prediction_parameters(freq, prediction_length)\n",
+    "list_of_df = predictor.predict(violation_list_training[:5])\n",
+    "actual_data = violation_list[:5]\n",
+    "for k in range(len(list_of_df)):\n",
+    "    plt.figure(figsize=(12,6), dpi=75, facecolor='w')\n",
+    "    plt.ylabel('Violations')\n",
+    "    plt.xlabel('Date')\n",
+    "    plt.title('Chicago Speed Camera Violations:' + unique_addresses[k])\n",
+    "    actual_data[k][-prediction_length-context_length:].plot(label='target')\n",
+    "    p10 = list_of_df[k]['0.1']\n",
+    "    p90 = list_of_df[k]['0.9']\n",
+    "    plt.fill_between(p10.index, p10, p90, color='y', alpha=0.5,label='80% confidence interval')\n",
+    "    list_of_df[k]['0.5'].plot(label='prediction median')\n",
+    "    plt.legend()\n",
+    "    plt.show()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "\n",
+    "### Clean-up\n",
+    "\n",
+    "At the end of this exercise, delete the realtime predictor endpoint to avoid accumulating charges for realtime predictor endpoint from your account."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predictor.delete_endpoint(endpoint_name)"
+   ]
+  }
+ ],
+ "metadata": {
```

Comment:
add the copyright license to the metadata. example: https://github.com/awslabs/amazon-sagemaker-examples/blame/master/aws_sagemaker_studio/frameworks/keras_pipe_mode_horovod/keras_pipe_mode_horovod_cifar10.ipynb#L621

Reply:
yes, added. thx